sdks/go/pkg/beam/core/sdf/sdf.go

```diff
@@ -87,6 +87,10 @@ type RTracker interface {
 	// GetRestriction returns the restriction this tracker is tracking, or nil if the restriction
 	// is unavailable for some reason.
 	GetRestriction() interface{}
+
+	// IsBounded returns the boundedness of the current restriction. If the current restriction represents a
+	// finite amount of work, it should return sdf.Bounded. Otherwise, it should return sdf.Unbounded.
+	IsBounded() bool
```

Comment:
This would be a breaking change for anyone who implements the RTracker interface (which is also why you needed to update all existing Trackers). One way to make this non-breaking would be to introduce a new interface that you can compose with this one, something like:

```
type UnboundableRTracker interface {
   IsBounded() bool
}
```
(UnboundableRTracker might be a bad name, feel free to come up with something better 😄 ). Then whenever we want to check boundedness, we can call something like:

```
isRtrackerBound(tracker RTracker) {
   if uTracker, ok := tracker.(UnboundableRTracker); ok && uTracker.IsBounded() {
      return true
   }
   return false
}
```

Reply:
Note - this implicitly treats all existing restrictions as bounded (which IMO is the right thing since we're introducing new behaviors here for unbounded restrictions)